### PR TITLE
Added Heuristic Search Algorithm in Algoritms/Heuristic_Search.cpp

### LINKED ISSUE #339: Heuristic Search Algorithm in cpp
Heuristic Search Algorithms are problem-solving methods that use heuristics—rules of thumb or strategies—to guide the search process toward finding optimal or satisfactory solutions more efficiently than traditional algorithms. These algorithms are particularly useful in complex problem spaces where exhaustive search is impractical.

### PR DESCRIPTION
**Pull request by Heet-Patel-5304**
# Description:

Added A* Search Algorithm based on Heuristic Search using C++

This pull request introduces an implementation of the A search algorithm* using a class-based approach in C++. The algorithm efficiently finds the shortest path in a grid by combining the actual cost from the start (g) and an estimated cost to the goal (h) using the Manhattan distance as the heuristic.

Key Highlights:
Point Class: A class to represent grid points, including coordinates (x, y), cost from the start (g), and heuristic value (h). It also includes an overloaded comparison operator to prioritize points based on their f = g + h value.
Priority Queue: Utilizes a priority_queue to explore the most promising paths first by always expanding the point with the lowest f value.
Heuristic Function: The Manhattan distance heuristic is used to estimate the distance between two points in the grid.
Validation: The isValid function ensures that moves remain within the grid boundaries and avoids obstacles.
How it works:
The algorithm starts at a given point (0,0) and explores neighboring points, calculating the cost and heuristic for each.
The process continues until the goal point (4,4) is reached or all possible paths are exhausted.
Example Usage:

Fixes #339

### CODE REVIEW

**Comment by Heet-Patel-5304** [comment]
I am looking forward to addressing any improvements that may arise from this implementation and am eager to contribute to other tasks as well. Please feel free to reach out if there are any questions or further enhancements needed.

File: README.md [hunk 1]
```diff
@@ -52,6 +52,7 @@
 │   │   ├── Hash Table Quadratic Probing.txt
 │   │   ├── HashTableLinearProbingC++.txt
 │   │   └── TwoSum.c++
+│   ├── Heuristic_Search.cpp
```
Comment (Kavya-24) [inline README.md:55]: You do not need to add this in README.md or project tree structure
That will be done automatically.

File: project_tree_structure.txt [hunk 2]
```diff
@@ -28,6 +28,7 @@
 │   │   ├── Hash Table Quadratic Probing.txt
 │   │   ├── HashTableLinearProbingC++.txt
 │   │   └── TwoSum.c++
+│   ├── Heuristic_Search.cpp
```
Comment (Kavya-24) [inline project_tree_structure.txt:31]: You do not need to add this in README.md or project tree structure
That will be done automatically on merge.

File: Algorithms/Heuristic_Search.cpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
+#include <bits/stdc++.h>
```
Comment (Kavya-24) [inline Algorithms/Heuristic_Search.cpp:1]: Hi @Heet-Patel-5304 , Thanks for contributing to this repository.
We do have some heuristic based search in this folder. Requesting you to move the CPP code to the same folder
https://github.com/Kavya-24/Snippets/tree/master/Artificial%20Intelligence/Informed%20Search/A%20Star
Thanks & happy coding !

**Review by Kavya-24 — COMMENTED** [review]
Some minor comments to be addressed.

File: project_tree_structure.txt (hunk 2 above)
Comment (Heet-Patel-5304) [reply project_tree_structure.txt]: It automatically updates README.md OR project tree structure. My new PR has conflict issues. Kindly help me to resolve them.

File: Algorithms/Heuristic_Search.cpp (hunk 3 above)
Comment (Heet-Patel-5304) [reply Algorithms/Heuristic_Search.cpp:1]: The File has been successfully moved to destined location. Looking forward to addressing any improvements.

**Pull request closed without merging**